src/services/codefixes/fixStrictClassInitialization.ts

```diff
@@ -109,14 +109,8 @@ namespace ts.codefix {
     }
 
     function getDefaultValueFromType (checker: TypeChecker, type: Type): Expression | undefined {
-        if (type.flags & TypeFlags.String) {
-            return createLiteral("");
-        }
-        else if (type.flags & TypeFlags.Number) {
-            return createNumericLiteral("0");
-        }
-        else if (type.flags & TypeFlags.Boolean) {

```

Comment:
Also have to change this since a `foo | boolean` union doesn't decompose to something containing `boolean`, it decomposes to boolean literal types.